File: java/lib/src/main/java/com/svix/BackgroundTaskListOptions.java
package com.svix;

import com.svix.models.BackgroundTaskStatus;
import com.svix.models.BackgroundTaskType;
import com.svix.models.Ordering;

public class BackgroundTaskListOptions extends ListOptions {
    private BackgroundTaskStatus status;
    private BackgroundTaskType task;
    private Ordering order;

    public void setOrder(final Ordering order) {
        this.order = order;
    }

    public Ordering getOrder() {
        return this.order;
    }

    public void setStatus(final BackgroundTaskStatus status) {
        this.status = status;
    }

    public BackgroundTaskStatus getStatus() {
        return this.status;
    }

    public void setTask(final BackgroundTaskType task) {
        this.task = task;
    }

    public BackgroundTaskType getTask() {
        return this.task;
    }
}

File: csharp/Svix/Models/BackgroundTaskListOptions.cs
using System;
using System.Collections.Generic;
using Svix.Model;

namespace Svix.Models
{
    public sealed class BackgroundTaskListOptions : ListOptions
    {
        public BackgroundTaskStatus? Status { get; set; }

        public BackgroundTaskType? Task { get; set; }
    }
}